src/main/scala/cromwell/engine/workflow/CallMetadataBuilder.scala

```diff
@@ -36,11 +34,14 @@ object CallMetadataBuilder {
   private type ExecutionMapTransformer = ExecutionMap => ExecutionMap
 
   object BackendValues {
-    def extract(job: Any): BackendValues = {
-      job match {
-        case ji: LocalJob => BackendValues("Local")
-        case ji: JesJob => BackendValues("JES", jobId = ji.jesRunId, status = ji.jesStatus)
-        case ji: SgeJob => BackendValues("SGE", jobId = ji.sgeJobNumber map { _.toString })
+    // FIXME needs to traverse the list of pluggable backends rather than hardcoding a list.
+    def extract(infos: ExecutionInfosByExecution): BackendValues = {
+      def extractValue(key: String): Option[String] = infos.executionInfos collectFirst { case i if i.key == key => i.value } flatten
+
+      infos.execution.backendType match {
+        case "Local" => BackendValues("Local", jobId = extractValue("PID"))
+        case "JES" => BackendValues("JES", jobId = extractValue("JES_RUN_ID"), status = extractValue("JES_STATUS"))
+        case "SGE" => BackendValues("SGE", jobId = extractValue("SGE_JOB_NUMBER"))
```

Comment:
I do like this idea, but it's both orthogonal to my changes and alters the metadata response in a way the artists formerly known as Blue might not appreciate.  I'll make a new ticket for this.